workitem/expression_compiler_blackbox_test.go

```diff
@@ -200,6 +200,25 @@ func TestIsNull(t *testing.T) {
 	expect(t, c.IsNull("SpaceID"), `(`+workitem.Column(wiTbl, "space_id")+` IS NULL)`, []interface{}{}, nil)
 }
 
+func TestChild(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	defJoins := workitem.DefaultTableJoins()
+	t.Run("iteration", func(t *testing.T) {
+		j := *defJoins["iteration"]
+		j.Active = true
+		e := "(uuid(\"work_items\".fields->>'system.iteration') IN (\n\t\t\t\tSELECT iter.id\n\t\t\t\t\tWHERE\n\t\t\t\t\t\t(SELECT j.path\n\t\t\t\t\t\t\tFROM iterations j\n\t\t\t\t\t\t\tWHERE j.id = ?\n\t\t\t\t\t\t) @> iter.path\n\t\t\t\t\t\t\t  ))"
+		expect(t, c.Child(c.Field("system.iteration"), c.Literal("abcd")), e, []interface{}{"abcd"}, []*workitem.TableJoin{&j})
+	})
+	t.Run("area", func(t *testing.T) {
+		j := *defJoins["area"]
+		j.Active = true
+		e := "(uuid(\"work_items\".fields->>'system.area') IN (\n\t\t\t\tSELECT iter.id\n\t\t\t\t\tWHERE\n\t\t\t\t\t\t(SELECT j.path\n\t\t\t\t\t\t\tFROM areas j\n\t\t\t\t\t\t\tWHERE j.id = ?\n\t\t\t\t\t\t) @> iter.path\n\t\t\t\t\t\t\t  ))"
```

Comment:
Why does a join on `area` have `iter.path`/`iter.id` in it?

Reply:
Good catch! Fixed in 761d3dc48661bf1b68caf211620e782ec5c8cbd1